pyproject.toml

```diff
@@ -1,7 +1,59 @@
+[project]
+name = "jf_agent"
+version = "0.0.1"
+description = "An agent that can run on-premise to download and send data to Jellyfish."
+authors = [
+    {name = "jellyfish-oss", email = "oss@jellyfish.co"},
+]
+dependencies = [
+    "urllib3>=1.26.5",
+    "pyyaml>=4.21b",
+    "jira<4,>=3.1.1",
+    "tqdm>=4.66.2",
+    "stashy",
+    "dateparser",
+    "jsonstreams",
+    "psutil",
+    "python-gitlab",
+    "click~=8.0.4",
+    "requests>=2.31.0",
+    "python-dotenv>=1.0.0",
+    "jf-ingest==0.0.105",
+    "structlog>=24.4.0",
+]
+requires-python = ">=3.10"
+readme = "README.md"
+license = {text = "MIT"}
+
+[build-system]
+requires = ["pdm-backend"]
+build-backend = "pdm.backend"
+
+[tool.pdm]
+[[tool.pdm.source]]
+url = "https://pypi.python.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[tool.pdm.dev-dependencies]
+dev = [
+    "requests-mock",
+    "pytest>=7.4.0",
+    "black==23.12.0",
+    "isort==5.12.0",
```

Comment:
Is there a compatibility matrix for black + others... what issues did you run into that keeps us from using the latest?

Reply:
Updated